oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java

```diff
@@ -175,7 +176,10 @@ public SampleFamily min(List<String> by) {
     }
 
     public SampleFamily avg(List<String> by) {
-        ExpressionParsingContext.get().ifPresent(ctx -> ctx.aggregationLabels.addAll(by));
+        ExpressionParsingContext.get().ifPresent(ctx -> {
+            ctx.aggregationLabels.addAll(by);
+            ctx.downsampling = DownsamplingType.AVG;
```

Comment:
AVG seems default in `ExpressionParsingContext#create`. Is this for making codes more clear or functional need?

Reply:
> AVG seems default in `ExpressionParsingContext#create`. Is this for making codes more clear or functional need?

`avg` being default had no problem before since it was only one function, now since I added another one, if the expression contains multiple function, and the `avg` being last one, it may be wrong.  For example, say the expression is `log_count.sum(by: ['service', 'instance']).avg(['level'])`, (this expression may make no sense, just for demonstration), the final metrics should be aggregated by `avg` instead of `sum`.

Reply:
Got it. A suggestion, could you use this PR to format `SampleFamily` a little better? Such as put `avg` in this way
```
    public SampleFamily avg(List<String> by) {
        ExpressionParsingContext.get().ifPresent(ctx -> ctx.aggregationLabels.addAll(by));
        if (this == EMPTY) {
            return EMPTY;
        }
        if (by == null) {
            double result = Arrays.stream(samples).mapToDouble(s -> s.value).average().orElse(0.0D);
            return SampleFamily.build(this.context, newSample(ImmutableMap.of(), samples[0].timestamp, result));
        }
        return SampleFamily.build(
            this.context,
            Arrays.stream(samples)
                  .map(sample -> 
                           Tuple.of(
                               by.stream()
                                 .collect(toImmutableMap(
                                     labelKey -> labelKey, 
                                     labelKey -> sample.labels.getOrDefault(labelKey, ""))
                                 ), 
                               sample)
                  )
                  .collect(groupingBy(Tuple2::_1, mapping(Tuple2::_2, toList())))
                  .entrySet().stream()
                  .map(entry -> 
                           newSample(entry.getKey(), 
                                     entry.getValue().get(0).timestamp, 
                                     entry.getValue().stream().mapToDouble(s -> s.value).average().orElse(0.0D)))
                  .toArray(Sample[]::new)
        );
    }
```

Reply:
Including several other methods. The codes mixed Tuple, it is hard to read.